### ISSUE TITLE
Is there a way to use this with rehype-highlight?

### ISSUE THREAD

**Issue by joostdecock** [remcohaszing/remark-mdx-code-meta#5]
Hi there,
Thanks for this, it works great for passing down the metadata as props, but I have hit a problem: It breaks syntax highlighting with `rehype-highlight`.

```js
import { compile } from '@mdx-js/mdx'
import codeMeta from 'remark-mdx-code-meta'
import rehypeHighlight from 'rehype-highlight'

compile(md, {
  outputFormat: 'function-body',
  remarkPlugins: [
    codeMeta, // adding this breaks syntax highlighting
  ],
  rehypePlugins: [
    [rehypeHighlight, { plainText: ['http'] }],
  ]
})
```
Result with your plugin:
![Screenshot from 2022-10-05 04-17-24](https://user-images.githubusercontent.com/1708494/193965162-99cbebb0-6f2e-43a8-9bd3-dcd9fd08dba3.png)

Result without your plugin:
![Screenshot from 2022-10-05 04-16-48](https://user-images.githubusercontent.com/1708494/193965163-d9b86f5f-c08b-4ce6-baf1-8e601ac5f9ae.png)

If you had any pointers on how to make this plugin play nice with [rehype-highlight](https://github.com/rehypejs/rehype-highlight) I would very much appreciate hearing about them :pray: 

thank you
joost

**Comment by remcohaszing** [comment]
Ideally this would just work, but I see why it doesn’t.

This plugin transforms markdown code blocks into MDX elements, skipping all rehype transformations. To fix this issue, the syntax highlighter would need to run before this plugin. That means this plugin will need to become a rehype plugin instead of a remark plugin. I would love to fix this, but at the moment I’m working on some other issues that will take longer.

The project I use this with uses client side syntax highlighting instead of build time. As a workaround you could use that for now.

TL;DR I would love to fix it, but not right now.

**Comment by FatehAK** [comment]
I'm facing the same issue, I'm trying to use it along with Astro's shiki highlighter. The syntax highlighting is lost when I use this plugin. Could you please help with this if you get time or guide on how to implement this?

**Comment by remcohaszing** [comment]
I just released [`rehype-mdx-code-props`](https://github.com/remcohaszing/rehype-mdx-code-props) which should be used instead.
